server/src/test/java/org/elasticsearch/gateway/PersistedClusterStateServiceTests.java

```diff
@@ -490,94 +284,13 @@ public void testFailsIfGlobalMetadataIsMissing() throws IOException {
             }
 
             final String message = expectThrows(IllegalStateException.class,
-                () -> newPersistedClusterStateService(nodeEnvironment).loadBestOnDiskState()).getMessage();
+                () -> newPersistedClusterStateService(nodeEnvironment).loadOnDiskState()).getMessage();
             assertThat(message, allOf(containsString("no global metadata found"), containsString(brokenPath.toString())));
         }
     }
 
-    public void testFailsIfGlobalMetadataIsDuplicated() throws IOException {
```

Comment:
This test is still meaningful, although we'll need to work out a more creative way to duplicate that document.

Reply:
Here is the same test but using two distinct `NodeEnvironment` objects to get the metadata indices that we mash together. This works on `master` as it stands today, obviously needs adjusting for the single-data-path case.

```
    public void testFailsIfGlobalMetadataIsDuplicated() throws IOException {
        // if someone attempted surgery on the metadata index by hand, e.g. deleting broken segments, then maybe the global metadata
        // is duplicated

        final Path[] dataPaths1 = createDataPaths();
        final Path[] dataPaths2 = createDataPaths();

        try (NodeEnvironment nodeEnvironment1 = newNodeEnvironment(dataPaths1);
             NodeEnvironment nodeEnvironment2 = newNodeEnvironment(dataPaths2)) {
            try (Writer writer1 = newPersistedClusterStateService(nodeEnvironment1).createWriter();
                 Writer writer2 = newPersistedClusterStateService(nodeEnvironment2).createWriter()) {
                final ClusterState oldClusterState = loadPersistedClusterState(newPersistedClusterStateService(nodeEnvironment1));
                final long newVersion = randomLongBetween(1L, Long.MAX_VALUE);
                final ClusterState newClusterState = ClusterState.builder(oldClusterState).version(newVersion).build();
                writeState(writer1, 0L, newClusterState, oldClusterState);
                writeState(writer2, 0L, newClusterState, oldClusterState);
            }

            final Path brokenPath = randomFrom(nodeEnvironment1.nodeDataPaths());
            final Path dupPath = randomFrom(nodeEnvironment2.nodeDataPaths());
            try (Directory directory = new SimpleFSDirectory(brokenPath.resolve(PersistedClusterStateService.METADATA_DIRECTORY_NAME));
                 Directory dupDirectory = new SimpleFSDirectory(dupPath.resolve(PersistedClusterStateService.METADATA_DIRECTORY_NAME))) {
                try (IndexWriter indexWriter = new IndexWriter(directory, new IndexWriterConfig())) {
                    indexWriter.addIndexes(dupDirectory);
                    indexWriter.commit();
                }
            }

            final String message = expectThrows(IllegalStateException.class,
                () -> newPersistedClusterStateService(nodeEnvironment1).loadBestOnDiskState()).getMessage();
            assertThat(message, allOf(containsString("duplicate global metadata found"), containsString(brokenPath.toString())));
        }
    }
```